Error when deleting Object Instances

When deleting a class in the data model, there are two errors ("OdTreeWalker.js:83 Error: element <om:Object id="Object_1v21jng" /> referenced by <om:Link id="Link_1l95ocq" />#targetRef not yet drawn" and "failed to import <om:Link id="Link_1maeft5" />") when an object instance of this class in the Objective Model will be deleted too because of the deletion in the data model.

1. What is the expected behaviour: Already has the expected behavior, but we don´t want to have an error in the console.
3. If not VERY clear: Why is this a problem

## Steps For Reproduction

1. go to Objective Model and in Objective
2. build two Object Instances that are linked
3. delete one of the corresponding classes from the Object Instances in the data model --> error is thrown

--- a/app/lib/objectivemodeler/OmModeler.js
+++ b/app/lib/objectivemodeler/OmModeler.js
@@ -325,5 +325,5 @@ OmModeler.prototype.isObjectOfDeletedClass = function (clazz, element) {
 }
 
 OmModeler.prototype.isLinkConnectedToObjectOfDeletedClass = function (clazz, element) {
-    return is(element, 'om:Link') && clazz.id && ((element.sourceRef?.classRef.id === clazz.id) || (element.targetRef?.classRef.id === clazz.id));
+    return is(element, 'om:Link') && clazz.id && ((element.sourceRef?.classRef?.id === clazz.id) || (element.targetRef?.classRef?.id === clazz.id));
 }

--- a/app/lib/objectivemodeler/OmModeler.js
+++ b/app/lib/objectivemodeler/OmModeler.js
@@ -254,10 +254,11 @@ OmModeler.prototype.handleClassRenamed = function (clazz) {
 OmModeler.prototype.handleClassDeleted = function (clazz) {
     let objectives = this._definitions.get('rootElements');
     objectives.forEach(objective => {
-        let objects = objective.get('boardElements');
-        for (let i = 0; i < objects.length; i++) {
-            if (is(objects[i], 'om:Object') && clazz.id && objects[i].classRef?.id === clazz.id) {
-                objects.splice(i, 1);
+        let elements = objective.get('boardElements');
+
+        for (let i = 0; i < elements.length; i++) {
+            if (this.isObjectOfDeletedClass(clazz, elements[i]) || this.isLinkConnectedToObjectOfDeletedClass(clazz, elements[i])) {
+                elements.splice(i, 1);
                 i--;
             }
         }
@@ -319,3 +320,10 @@ OmModeler.prototype.getObjectiveByReference = function (objectiveReference) {
     }
 }
 
+OmModeler.prototype.isObjectOfDeletedClass = function (clazz, element) {
+    return is(element, 'om:Object') && clazz.id && element.classRef?.id === clazz.id;
+}
+
+OmModeler.prototype.isLinkConnectedToObjectOfDeletedClass = function (clazz, element) {
+    return is(element, 'om:Link') && clazz.id && ((element.sourceRef?.classRef.id === clazz.id) || (element.targetRef?.classRef.id === clazz.id));
+}

